### ISSUE TITLE
See how to improve multiline string with qm-interpolated-string

### ISSUE THREAD

**Issue by rexim** [tsoding/boids#34]
https://github.com/tsoding/boids/blob/dc4970533c1cd6c0d413c630ecffaa3ea808eeb1/test/TestDataSpec.hs#L92-L104

https://www.stackage.org/package/qm-interpolated-string

Suggested by @unclechu

**Comment by unclechu** [comment]
Keep in mind that this package not in any of Stack's LTS yet, you should explicitly specify it ([an example](https://github.com/unclechu/xlib-keys-hack/blob/f83d90fd4ceba640957956484a17453bdadf7aa7/stack.yaml#L11)).

**Comment by unclechu** [comment]
```haskell
where xmlData = unlines [ "<svg xmlns='http://www.w3.org/2000/svg'>"
                        , "  <g id='outer'>"
                        , "    <circle cx='326' cy='155' />"
                        , "    <circle cx='478' cy='419' />"
                        , "    <circle cx='107' cy='449' />"
                        , "    <g id='inner'>"
                        , "      <circle cx='102' cy='152' />"
                        , "      <circle cx='327' cy='246' />"
                        , "      <circle cx='444' cy='358' />"
                        , "    </g>"
                        , "  </g>"
                        , "</svg>"
                        ]
```
Could be replaced with:
```haskell
{-# LANGUAGE QuasiQuotes #-}
import Text.InterpolatedString.QM (qm)

-- ...

where xmlData = unlines [qm| <svg xmlns='http://www.w3.org/2000/svg'>
                               <g id='outer'>
                                 <circle cx='326' cy='155' />
                                 <circle cx='478' cy='419' />
                                 <circle cx='107' cy='449' />
                                 <g id='inner'>
                                   <circle cx='102' cy='152' />
                                   <circle cx='327' cy='246' />
                                   <circle cx='444' cy='358' />
                                 </g>
                               </g>
                             </svg>
                           |]
```

**Comment by unclechu** [comment]
`QuasiQuotes` extension could be used to build a DSL.
Look at this for example: https://www.stackage.org/lts-8.4/package/inline-c-0.5.6.1

**Comment by rexim** [comment]
@unclechu feel free to create a pull request. :)

**Comment by unclechu** [comment]
@rexim [PR](https://github.com/tsoding/boids/pull/36)

**Comment by rexim** [comment]
Resolved in 961effc4d56368024693383869f2e4bfd162307b
